tests/serverpod_test_server/test_integration/test_tools/serverpod_test_tools.dart

```diff
@@ -5148,23 +5148,25 @@ class _MapParametersEndpoint {
           createSessionCallback: (_) => _localUniqueSession,
           endpointPath: 'mapParameters',
           methodName: 'returnIntIntMap',
-          parameters: _i1.testObjectToJson({'map': map}),
+          parameters: _i1.testObjectToJson({'map': map.isEmpty ? [] : map}),
           serializationManager: _serializationManager,
         );
-        var _localReturnValue = await (_localCallContext.method.call(
-          _localUniqueSession,
-          _localCallContext.arguments,
-        ) as _i3.Future<Map<int, int>>);
+        var _localReturnValue = await _localCallContext.method
+            .call(
+              _localUniqueSession,
+              _localCallContext.arguments,
+            )
+            .then((map) => _i17.Protocol().deserialize<Map<int, int>>(map));
```

Comment:
Here is an example of the code-gen change, reverting the partially mapped map back.